packages/core/admin/_internal/node/develop.ts

```diff
@@ -24,6 +27,41 @@ interface DevelopOptions extends CLIContext {
   watchAdmin?: boolean;
 }
 
+const cleanupDistDirectory = async ({
+  tsconfig,
+  logger,
+  timer,
+}: Pick<DevelopOptions, 'tsconfig' | 'logger'> & { timer: TimeMeasurer }) => {
+  const distDir = tsconfig?.config?.options?.outDir;
+
+  if (!distDir || !(await fs.pathExists(distDir))) {
+    return;
+  }
+  const timerName = 'cleaningDist' + Date.now();
+  timer.start(timerName);
+  const cleaningSpinner = logger.spinner(`Cleaning dist dir ${distDir}`).start();
+
+  try {
+    const dirContent = await fs.readdir(distDir);
+    const validFilenames = dirContent
+      // Ignore the admin build folder
+      .filter((filename) => filename !== 'build');
+    for (const filename of validFilenames) {
+      await fs.remove(path.resolve(distDir, filename));
+    }
+    throw new Error('asdf');
```

Comment:
this feels like an oversight?

Reply:
Hah, yes, currently working in it as we speak, give me 10 minutes